app/src/lib/git/commit.ts

```diff
@@ -46,20 +56,30 @@ export async function createCommit(
  */
 export async function createMergeCommit(
   repository: Repository,
-  files: ReadonlyArray<WorkingDirectoryFileChange>
+  files: ReadonlyArray<WorkingDirectoryFileChange>,
+  manualResolutions: ReadonlyMap<string, ManualConflictResolution> = new Map()
 ): Promise<string | undefined> {
   // Clear the staging area, our diffs reflect the difference between the
   // working directory and the last commit (if any) so our commits should
   // do the same thing.
   try {
     await unstageAll(repository)
-    await stageFiles(repository, files)
+
+    // apply manual conflict resolutions
+    for (const [path, resolution] of manualResolutions) {
+      const file = files.find(f => f.path === path)
+      if (file !== undefined) {
```

Comment:
Do you think we should add some debug tracing in here to catch if a `manualResolutions` entry doesn't exist in the `files` list?